### PR TITLE
Fix zstd decomporession

### PR DESCRIPTION
**Pull request by zendai**
#### Description

This is a fix to issue #6914 .

#### Checklist

 - [x] I have updated tests where applicable.
 - [x] I have added an entry to the CHANGELOG.

### CODE REVIEW

File: test/mitmproxy/net/test_zstd.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from mitmproxy.net.encoding import decode_zstd
```
Comment (mhils) [inline test/mitmproxy/net/test_zstd.py:1]: Please add this to `test_encoding.py`. We want to keep parity between source file and test file name. :)

File: mitmproxy/net/encoding.py [hunk 2]
```diff
@@ -174,12 +174,7 @@ def decode_zstd(content: bytes) -> bytes:
     if not content:
         return b""
     zstd_ctx = zstd.ZstdDecompressor()
-    try:
-        return zstd_ctx.decompress(content)
-    except zstd.ZstdError:
-        # If the zstd stream is streamed without a size header,
-        # try decoding with a 10MiB output buffer
-        return zstd_ctx.decompress(content, max_output_size=10 * 2**20)
+    return zstd_ctx.stream_reader(BytesIO(content)).read()
```
Comment (mhils) [inline mitmproxy/net/encoding.py:177]: Why does this work without passing `read_across_frames=True`?

According to https://python-zstandard.readthedocs.io/en/stable/decompressor.html#zstandard.ZstdDecompressor.stream_reader, it's false by default. Is this a bug in the underlying library, or why are things working here?

**Review by mhils — COMMENTED** [review]
Thanks! 🍰

File: test/mitmproxy/net/test_zstd.py (hunk 1 above)
Comment (zendai) [reply test/mitmproxy/net/test_zstd.py:1]: Sure! Thanks for the guidance.

File: mitmproxy/net/encoding.py (hunk 2 above)
Comment (zendai) [reply mitmproxy/net/encoding.py:177]: Fair question, I don't know. I can add explicitly `read_across_frames=True` and test to ensure it works. Fair point, I'd sleep better if the code would align with the documentation. Let me play with this a little, will come back.

File: test/mitmproxy/net/test_zstd.py (hunk 1 above)
Comment (zendai) [reply test/mitmproxy/net/test_zstd.py]: Moved the test to test_encoding.py.

File: mitmproxy/net/encoding.py (hunk 2 above)
Comment (zendai) [reply mitmproxy/net/encoding.py]: So, I tested it and it works both with explicit False and True as well. I tracked down where this flags comes into play during Decompress' stream_reader().[read()](https://github.com/indygreg/python-zstandard/blob/518ade454f41ffbad8b9b542f6938cd308653ba2/zstandard/backend_cffi.py#L3288 ), it's [here ](https://github.com/indygreg/python-zstandard/blob/518ade454f41ffbad8b9b542f6938cd308653ba2/zstandard/backend_cffi.py#L3282) and it's part of an OR condition:

```python
        # Emit data if there is data AND either:
        # a) output buffer is full (read amount is satisfied)
        # b) we're at end of a frame and not in frame spanning mode
        return out_buffer.pos and (
            out_buffer.pos == out_buffer.size
            or zresult == 0
            and not self._read_across_frames
        )
```

I'm not going to pretend that I fully understand this part, I have some assumptions but I'm not familiar with zstd to this depth to say anything here with certainty. Still, I added the `read_across_frames=True` to the PR as based on my tests it works as well and based on the documentation it makes sense to have it.

**Review by mhils — APPROVED** [review]

**Pull request merged**